parlai/tasks/dialogue_safety/agents.py

```diff
@@ -0,0 +1,533 @@
+            # loop and add other rounds
+            for i in range(self.round - 1):
+                rnd = str(i + 1)
+                for x in ['good', 'bad']:
+                    self.data += self.data_dump[d_type][rnd][x]
+
+        # add data from current round
+        for x in ['good', 'bad']:
+            self.data += self.data_dump[d_type][str(self.round)][x]
+
+        self.fixed_random.shuffle(self.data)
+
+    def get(self, episode_idx, entry_idx):
+        return Message(self.data[episode_idx])
+
+    def share(self):
+        shared = super().share()
+        shared['data'] = self.data
+        return shared
+
+
+class MultiturnTeacher(FixedDialogTeacher):
+    """
+    Data from the multi-turn adversarial collection described in the paper
+    `Build it Break it Fix it for Dialogue Safety: Robustness from
+    Adversarial Human Attack` (<https://arxiv.org/abs/1908.06083>)
+
+    To see data containing mulit-turn conversations, try running
+    `python examples/display_data.py -t dialogue_safety:multiturn`.
+
+    Run the above command with the flag `--single-turn True` to only see the
+    single turn data.
+    """
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        parser = parser.add_argument_group('Multiturn Safety Teacher Args')
+        parser.add_argument(
+            '--single-turn',
+            type='bool',
+            default=False,
+            help='only include the single turn data and not the context info',
+        )
+
+    def __init__(self, opt, shared=None):
+        build(opt['datapath'])  # download the data
+        self.opt = opt
+        self.data_path = os.path.join(
+            opt['datapath'], 'dialogue_safety', MULTI_TURN_DATA
+        )
+
+        self.fixed_random = random.Random(42)
+        self.single_turn = opt['single_turn']
+
+        if shared and 'data' in shared:
+            self.data = shared['data']
+        else:
+            self._setup_data(opt['datatype'])
+
+        super().__init__(opt, shared)
+        self.reset()
+
+    def num_episodes(self):
+        return len(self.data)
+
+    def num_examples(self):
+        return len(self.data)
+
+    def share(self):
+        shared = super().share()
+        shared['data'] = self.data
+        return shared
+
+    def _setup_data(self, datatype):
+        dt = datatype.split(':')[0]
+        self.all_data = json.load(open(self.data_path, 'rb'))
+        data = self.all_data[dt]
+        if self.single_turn:
+            # remove
+            new_data = []
+            for datum in data:
+                datum['text'] = datum['text'].split('\n')[-1]
+                new_data.append(datum)
+            self.data = new_data
+        else:
+            self.data = data
+
+    def get(self, episode_idx, entry_idx):
+        return Message(self.data[episode_idx])
+
+
+class WikiToxicCommentsTeacher(FixedDialogTeacher):
+    """
+    Dataset of comments from Wikipedia's Talk page edits. Taken from
```

Comment:
might be good to also specify that you're converting it to a binary classification task